lib/src/test/java/com/basistheory/android/view/CardVerificationCodeElementTests.kt

```diff
@@ -96,4 +94,78 @@ class CardVerificationCodeElementTests {
 
         verify(exactly = 1) { cardNumberElement.addChangeEventListener(any()) }
     }
+
+    @Test
+    fun `emits a change event when mask increases from 3 to 4 chars`() {
+        val changeEvents = mutableListOf<ChangeEvent>()
+        cvcElement.addChangeEventListener { changeEvents.add(it) }
+
+        cardNumberElement.setText("4242424242424242")
+        cvcElement.cardNumberElement = cardNumberElement
+        cvcElement.setText("123")
+
+        expectThat(changeEvents).single().and {
+            get { isValid }.isTrue()
+            get { isEmpty }.isFalse()
+            get { isComplete }.isTrue()
+        }
+
+        // change the card brand from visa to american-express
+        cardNumberElement.setText("378282246310005")
+
+        expectThat(cvcElement.getText()).isEqualTo("123") // value does not change
+        expectThat(changeEvents).hasSize(2).last().and {
+            get { isValid }.isFalse()
+            get { isEmpty }.isFalse()
+            get { isComplete }.isFalse()
+        }
+    }
+
+    @Test
+    fun `emits a change event when mask decreases from 4 to 3 chars`() {
+        val changeEvents = mutableListOf<ChangeEvent>()
+        cvcElement.addChangeEventListener { changeEvents.add(it) }
+
+        cardNumberElement.setText("378282246310005")
+        cvcElement.cardNumberElement = cardNumberElement
+        cvcElement.setText("1234")
+
+        expectThat(changeEvents).single().and {
+            get { isValid }.isTrue()
+            get { isEmpty }.isFalse()
+            get { isComplete }.isTrue()
+        }
+
+        // change the card brand from visa to american-express
+        cardNumberElement.setText("4242424242424242")
+
+        expectThat(cvcElement.getText()).isEqualTo("1234") // value does not change
```

Comment:
so it is their problem to make sure it is not tokenized like this since the element still have 4 digits right?